File: NumbersOfOnes/NumbersOfOnesv1.java
package NumbersOfOnes;

public class NumbersOfOnesv1 {
    public static void main(String[] args){
        System.out.println(countDigitOne(13)); // Output: 6
        System.out.println(countDigitOne(824883294)); // Output: 6
    }

    public static int countDigitOne(int n){
        if (n > 0 && n <= 9) {return 1;}
        if (n <= 0) {return 0;}
        int count = 0;
        for (int i = 1; i <= n; i++){
            int num = i;
            while (num > 0){
                if (num % 10 == 1){
                    count++;
                }
                num /= 10;
            }
        }
        return count;
    }
}

File: NumbersOfOnes/NumbersOfOnesv1.py
def countDigitOne(n):
    count, factor = 0, 1

    while factor <= n:
        lower = n % factor
        current = (n // factor) % 10
        higher = n // (factor * 10)

        if current == 0:
            count += higher * factor
        elif current == 1:
            count += higher * factor + lower + 1
        else:
            count += (higher + 1) * factor

        factor *= 10

    return count


n = int(input("Introduce a number: "))
num1 = countDigitOne(n)

if num1 == 1:
    print(f"{n} has a total of one 1.")
else:
    print(f"{n} has a total of {num1} ones.")
